public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -36,72 +46,221 @@ export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewStat
+      throw new Error('SceneDataLayers not found');
     }
 
     return data;
   }
 
+  public getDataLayer(editIndex: number): dataLayers.AnnotationsDataLayer {
+    const data = this.getSceneDataLayers();
+    const layer = data.state.layers[editIndex];
+
+    if (!(layer instanceof dataLayers.AnnotationsDataLayer)) {
+      throw new Error('AnnotationsDataLayer not found at index ' + editIndex);
+    }
+
+    return layer;
+  }
+
   public getAnnotationsLength(): number {
-    return this._dataLayers.filter((layer) => layer.topic === DataTopic.Annotations).length;
+    return this.getSceneDataLayers().state.layers.filter((layer) => layer.topic === DataTopic.Annotations).length;
   }
 
   public getDashboard(): DashboardScene {
     return this._dashboard;
   }
 
   public onNew = () => {
-    console.log('todo: onNew');
+    const newAnnotationQuery: AnnotationQuery = {
+      name: newAnnotationName,
+      enable: true,
+      datasource: getDataSourceRef(getDataSourceSrv().getInstanceSettings(null)!),
+      iconColor: 'red',
+    };
+
+    const newAnnotation = new DashboardAnnotationsDataLayer({
+      key: `annotations-${newAnnotationQuery.name}`,
+      query: newAnnotationQuery,
+      name: newAnnotationQuery.name,
+      isEnabled: Boolean(newAnnotationQuery.enable),
+      isHidden: Boolean(newAnnotationQuery.hide),
+    });
+
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    newAnnotation.activate();
+
+    //keep annotation layers together
+    layers.splice(this.getAnnotationsLength(), 0, newAnnotation);
+
+    data.setState({
+      layers,
+    });
+
+    this.setState({ editIndex: this.getAnnotationsLength() - 1 });
   };
 
   public onEdit = (idx: number) => {
-    console.log('todo: onEdit');
+    this.setState({ editIndex: idx });
+  };
+
+  public goBackToList = () => {
+    this.setState({ editIndex: undefined });
+  };
+
+  public onPreview = () => {
+    this.setState({ editIndex: undefined });
+
+    this._dashboard.stopUrlSync();
+
+    locationService.partial({ editview: null });
+
+    this._dashboard.startUrlSync();
   };
 
-  public onMove = (idx: number, direction: number) => {
-    console.log('todo: onMove');
+  public onMove = (idx: number, direction: MoveDirection) => {
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    const [layer] = layers.splice(idx, 1);
+    layers.splice(idx + direction, 0, layer);
+
+    data.setState({
+      layers,
+    });
   };
 
   public onDelete = (idx: number) => {
-    console.log('todo: onDelete');
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    layers.splice(idx, 1);
+
+    data.setState({
+      layers,
+    });
+  };
+
+  public onUpdate = (annotation: AnnotationQuery, editIndex: number) => {
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    const layer = layers[editIndex];
```

Comment:
```suggestion
    const layer = this.getDataLayer(editIndex);
```
Maybe we can use the layer-getter here, WDYT?